File: app/src/main/java/se/hactar/movieregister/helper/ImportMovie.java
package se.hactar.movieregister.helper;

import se.hactar.movieregister.model.Movie;
import timber.log.Timber;

public class ImportMovie {

    public static Movie parse(final String line) {
        // Example: Film 1¤4¤tt0372588¤DVD¤Team America: World Police¤2004
        if (!line.matches(".*¤.*¤.*¤.*¤.*¤.*")) {
            Timber.i("Ignoring line that does not match import pattern, line=" + line);
        }
        final String[] columns = line.split("¤");
        Movie movie = new Movie();
        movie.setContainer(get(columns, Columns.CONTAINER));
        movie.setIndex(get(columns, Columns.INDEX));
        movie.setImdbId(get(columns, Columns.IMDB_ID));
        movie.setType(get(columns, Columns.TYPE));
        movie.setName(get(columns, Columns.NAME));
        movie.setYear(get(columns, Columns.YEAR));
        return movie;
    }

    private static String get(final String[] columns, final ImportMovie.Columns column) {
        return column.ordinal() < columns.length ? columns[column.ordinal()] : "";
    }

    private enum Columns {CONTAINER, INDEX, IMDB_ID, TYPE, NAME, YEAR}
}

File: app/src/main/java/se/hactar/movieregister/helper/ImportMovie.kt
package se.hactar.movieregister.helper

import android.util.Log
import se.hactar.movieregister.model.Movie

object ImportMovie {

    private val TAG = ImportMovie::class.java.simpleName

    // TODO: Move to Movie.from()
    fun parse(line: String): Movie {
        // Example: A|1|tt0107048|DVD|Groundhog Day|1993|
        // Example: A|13|tt0080274|DVD|Shogun |1980|Disc 1
        if (!line.matches(".*\\|.*\\|.*\\|.*\\|.*\\|.*\\|.*".toRegex())) {
            Log.i(TAG, "Ignoring line that does not match import pattern, line=$line")
        }
        val columns = line.split("\\|".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        val movie = Movie().apply {
            container = get(columns, Columns.CONTAINER)
            index = get(columns, Columns.INDEX)
            imdbId = get(columns, Columns.IMDB_ID)
            discType = get(columns, Columns.DISC_TYPE)
            title = get(columns, Columns.TITLE)
            year = get(columns, Columns.YEAR)
            indexTitle = get(columns, Columns.INDEX_TITLE)
        }
        return movie
    }

    private operator fun get(columns: Array<String>, column: Columns): String {
        return if (column.ordinal < columns.size) columns[column.ordinal] else ""
    }

    private enum class Columns {
        CONTAINER, INDEX, IMDB_ID, DISC_TYPE, TITLE, YEAR, INDEX_TITLE
    }
}
